File: septunneuf-serveur/septunneuf-shop/src/main/java/fr/ebiron/septunneuf/shop/utils/RandomGenerators.java
package fr.ebiron.septunneuf.shop.utils;

import java.util.Random;

public class RandomGenerators {
    private static final Random RANDOM = new Random();

    public static int randomEggPrice() {
        return RANDOM.nextInt(5) + 1;
    }

    public static int randomMonsterPrice() {
        return RANDOM.nextInt(20) + 5;
    }

    public static int randomQuantity() {
        return RANDOM.nextInt(9) + 3;
    }

}

File: septunneuf-serveur/septunneuf-eggs/src/main/kotlin/fr/ebiron/septunneuf/eggs/utils/RandomGenerators.kt
package fr.ebiron.septunneuf.eggs.utils

import kotlin.random.Random
import kotlin.time.Duration
import kotlin.time.DurationUnit
import kotlin.time.toDuration

fun randomHexColor(): String {
    val randomValue = Random.nextInt(0xffffff + 1)
    return String.format("#%06x", randomValue)
}

fun randomIncubationTime(
    minIncubationTime: Duration = 30.toDuration(DurationUnit.SECONDS),
    maxIncubationTime: Duration = 5.toDuration(DurationUnit.MINUTES)
): Duration {
    require(minIncubationTime < maxIncubationTime) { "minIncubationTime must be less than maxIncubationTime" }

    val minSeconds = minIncubationTime.inWholeSeconds
    val maxSeconds = maxIncubationTime.inWholeSeconds
    val randomSeconds = Random.nextLong(minSeconds, maxSeconds + 1)

    return randomSeconds.toDuration(DurationUnit.SECONDS)
}